eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/Field.php

```diff
@@ -161,7 +169,7 @@ public function handle(
                     "A field of type '{$fieldTypeIdentifier}' is not searchable in the legacy search engine."
                 );
             }
-
+            $fieldsInfo['column'] = $fieldTypeIdentifier === 'ezkeyword' ? 'keyword' : $fieldsInfo['column'];
```

Comment:
This probably should be handled inside eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\KeywordConverter, but simple change in getIndexColumn() would mess up sorting functionality.